org.hl7.fhir.r4/src/main/java/org/hl7/fhir/r4/profilemodel/gen/PECodeGenerator.java

```diff
@@ -453,8 +450,10 @@ private void genSave(boolean isPrim, boolean isAbstract, String name, String sna
       w(save, "    tgt.clear(\""+fname+"\");");
       if (isList) {
         w(save, "    for ("+type+" item : "+name+") {");
-        if (isExtension) {
-          w(save, "      tgt.makeChild(\""+fname+"\").data().setProperty(\"value[x]\", item);");          
+        if (isExtension && !Strings.isNullOrEmpty(typeInfo.getUrl()) && typeInfo.getUrl().startsWith("http://hl7.org/fhir/StructureDefinition")) {
+          w(save, "      tgt.makeChild(\""+fname+"\").data().setProperty(\"value[x]\", item);");
+        } else if (isExtension) {
+          w(save, "      tgt.makeChild(\""+fname+"\").data().setProperty(\"value[x]\", item.getData());");
```

Comment:
@grahamegrieve I would really like to know if there's an alternative to invoke `getData` here.